homeassistant/components/overkiz/sensor.py

```diff
@@ -0,0 +1,418 @@
+        native_value=lambda value: str(value).capitalize(),
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.CORE_RSSI_LEVEL,
+        name="RSSI Level",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_value=lambda value: round(float(value)),
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.CORE_EXPECTED_NUMBER_OF_SHOWER,
+        name="Expected Number Of Shower",
+        icon="mdi:shower-head",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.CORE_NUMBER_OF_SHOWER_REMAINING,
+        name="Number of Shower Remaining",
+        icon="mdi:shower-head",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # V40 is measured in litres (L) and shows the amount of warm (mixed) water with a temperature of 40 C, which can be drained from a switched off electric water heater.
+    OverkizSensorDescription(
+        key=OverkizState.CORE_V40_WATER_VOLUME_ESTIMATION,
+        name="Water Volume Estimation at 40 °C",
+        icon="mdi:water",
+        native_unit_of_measurement=VOLUME_LITERS,
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.CORE_WATER_CONSUMPTION,
+        name="Water Consumption",
+        icon="mdi:water",
+        native_unit_of_measurement=VOLUME_LITERS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.IO_OUTLET_ENGINE,
+        name="Outlet Engine",
+        icon="mdi:fan-chevron-down",
+        native_unit_of_measurement=VOLUME_LITERS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.IO_INLET_ENGINE,
+        name="Inlet Engine",
+        icon="mdi:fan-chevron-up",
+        native_unit_of_measurement=VOLUME_FLOW_RATE_CUBIC_METERS_PER_HOUR,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.HLRRWIFI_ROOM_TEMPERATURE,
+        name="Room Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=TEMP_CELSIUS,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.IO_MIDDLE_WATER_TEMPERATURE,
+        name="Middle Water Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=TEMP_CELSIUS,
+    ),
+    OverkizSensorDescription(
+        key=OverkizState.CORE_FOSSIL_ENERGY_CONSUMPTION,
+        name="Fossil Energy Consumption",
+        device_class=SensorDeviceClass.ENERGY,
```

Comment:
We're missing a unit of measurement here. Energy sensors must use Wh, kWh or MWh.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes